hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpcengine/nio/NioBlockRequestScheduler.java

```diff
@@ -0,0 +1,291 @@
+                    reactor.wakeup();
+                });
+                break;
+            case BLK_REQ_OP_OPEN:
+                EXECUTOR.execute(() -> {
+                    try {
+                        req.channel = AsynchronousFileChannel.open(
+                                Path.of(req.file.path()),
+                                flagsToOpenOptions(req.flags));
+                        req.result = 0;
+                    } catch (IOException e) {
+                        req.exc = e;
+                    }
+
+                    cq.add(req);
+                    reactor.wakeup();
+                });
+                break;
+            case BLK_REQ_OP_CLOSE:
+                EXECUTOR.execute(() -> {
+                    try {
+                        req.getFile().channel.close();
+                        req.result = 0;
+                    } catch (IOException e) {
+                        req.exc = e;
+                    }
+
+                    cq.add(req);
+                    reactor.wakeup();
+                });
+                break;
+            case BLK_REQ_OP_FALLOCATE:
+                throw new RuntimeException("Not implemented yet.");
+            default:
+                throw new IllegalStateException("Unknown request type: " + req.opcode);
+        }
+    }
+
+    private OpenOption[] flagsToOpenOptions(int flags) {
+        int p = (flags & 3);
+        int supportedOpsExceptRW = (O_CREAT | O_TRUNC | O_DIRECT | O_SYNC);
+        int len = Integer.bitCount(flags & supportedOpsExceptRW) + p == O_RDWR ? 2 : 1;
+        // can be pooled
+        OpenOption[] opts = new OpenOption[len];
+        int i = 0;
+
+        if (p == O_RDONLY) {
+            opts[i++] = StandardOpenOption.READ;
+        } else if (p == O_WRONLY) {
+            opts[i++] = StandardOpenOption.WRITE;
+        } else if (p == O_RDWR) {
+            opts[i++] = StandardOpenOption.READ;
+            opts[i++] = StandardOpenOption.WRITE;
+        }
+
+        if ((flags & O_CREAT) != 0) {
+            opts[i++] = StandardOpenOption.CREATE;
+        }
+
+        if ((flags & O_TRUNC) != 0) {
+            opts[i++] = StandardOpenOption.TRUNCATE_EXISTING;
+        }
+
+        if ((flags & O_DIRECT) != 0) {
+            opts[i++] = ExtendedOpenOption.DIRECT;
+        }
+
+        if ((flags & O_SYNC) != 0) {
+            opts[i++] = StandardOpenOption.SYNC;
+        }
+
+        assert i == len;
+
+        return opts;
+    }
+
+    void complete(NioBlockRequest nioBlockRequest) {
+        if (nioBlockRequest.exc != null) {
+            nioBlockRequest.promise.completeExceptionally(nioBlockRequest.exc);
+        } else {
+            if (nioBlockRequest.opcode == BLK_REQ_OP_OPEN) {
+                nioBlockRequest.getFile().channel = nioBlockRequest.channel;
+            }
+            nioBlockRequest.promise.complete(nioBlockRequest.result);
+            handleMetrics(nioBlockRequest);
+        }
+
+        nioBlockRequest.rwFlags = 0;
```

Comment:
To improve readability, I would just call this 'request'. From the context it is already clear what it is.